src/Analysis/Codelyzer.Analysis.Build/WorkspaceBuilderHelper.cs

```diff
@@ -459,6 +461,40 @@ private EnvironmentOptions GetEnvironmentOptions(IProjectFile projectFile)
                 }
             }
 
+            try
+            {
+                MSBuildLocator.RegisterDefaults();
+            }
+            catch (System.InvalidOperationException)
+            {
+                var enviroment = System.Environment.CurrentDirectory;
+                string projectDirectory = Directory.GetParent(enviroment).Parent.FullName;
+                var proc = new Process
+                {
+                    StartInfo = new ProcessStartInfo
+                    {
+                        FileName = Path.Combine(projectDirectory, "vswhere.exe"),
+                        Arguments = "-latest -requires Microsoft.Component.MSBuild -find MSBuild\\**\\Bin\\MSBuild.exe",
+                        UseShellExecute = false,
+                        RedirectStandardOutput = true,
+                        CreateNoWindow = true,
+                        WorkingDirectory = ""
```

Comment:
Where does vswhere.exe come from? Is this going to be packaged with codelyzer?

Reply:
Yes. It is in the src/ directory. We can change the location to some other directory if required.